src/coordsim/simulation/flowsimulator.py

```diff
@@ -223,7 +236,15 @@ def process_flow(self, flow, sfc):
                              .format(flow.flow_id, current_sf, current_node_id, self.env.now))
                     # Remove the active flow from the SF after it departed the SF
                     metrics.remove_active_flow(flow, current_node_id, current_sf)
-                self.params.network.nodes[current_node_id]["remaining_cap"] += flow.dr
+
+                # Remove load from sf
+                self.params.network.nodes[current_node_id]['available_sf'][sf]['load'] -= flow.dr
```

Comment:
`asserts`, eg, here to check that the load is never below 0 could help to catch unintended errors in the load computations that are otherwise hard to see. (like we had them before)